importer/likutim/print_with_doc.go

```diff
@@ -0,0 +1,159 @@
+	"time"
+)
+
+type PrintWithDoc struct{}
+
+type printData struct {
+	cuUid    string
+	oName    string
+	filmDate time.Time
+	topic    string
+}
+
+func (c *PrintWithDoc) Run() {
+	mdb := c.openDB()
+	defer mdb.Close()
+
+	dir, err := ioutil.TempDir("/home/david", "temp_kitvei_makor")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+
+	cus, err := models.ContentUnits(mdb,
+		qm.Select("distinct on (\"content_units\".id) \"content_units\".*"),
+		qm.InnerJoin("files f ON f.content_unit_id = \"content_units\".id"),
+		qm.Where("type_id = ?", common.CONTENT_TYPE_REGISTRY.ByName[common.CT_KITEI_MAKOR].ID),
+		qm.Load("Files", "Tags", "Tags.TagI18ns"),
+	).All()
+	if err != nil {
+		log.Errorf("can't load units. Error: %s", err)
+	}
+
+	forPrint := make([]printData, 0)
+
+	for _, cu := range cus {
+
+		if err != nil {
+			log.Errorf("Error on parse date. Error: %s", err)
+		}
+		cuo, err := FindOrigin(mdb, cu.ID)
+		if err != nil {
+			log.Errorf("Error on looking for origins for unit Error: %s", err)
+		}
+		forPrint = append(forPrint, c.prepareForPrint(cu, cuo))
+	}
+	c.printToCSV(forPrint)
+}
+
+func (c *PrintWithDoc) prepareForPrint(cu, cuo *models.ContentUnit) printData {
+	var cuProps map[string]interface{}
+	err := json.Unmarshal(cu.Properties.JSON, &cuProps)
+	if err != nil {
+		log.Errorf("json.Unmarshal cu properties %d, Error: %s", cu.ID, err)
+	}
+	var film time.Time
+	if cuProps != nil {
+		film, err = time.Parse("2006-01-02", cuProps["film_date"].(string))
+		if err != nil {
+			log.Errorf("time.Parse cu %d film_date %s", cu.ID, cuProps["film_date"])
+		}
+	}
+
+	name := null.String{}
+	if cuo != nil {
+		for _, o := range cuo.R.ContentUnitI18ns {
+			if o.Language == "he" {
+				name = o.Name
+			}
+		}
+	}
+
+	label := null.String{}
+	for _, t := range cu.R.Tags {
+		for _, ttr := range t.R.TagI18ns {
+			if ttr.Language == "he" {
+				label = ttr.Label
+			}
+		}
+	}
+
+	return printData{
+		cuUid:    cu.UID,
+		oName:    name.String,
+		filmDate: film,
+		topic:    label.String,
+	}
+}
+
+func (c *PrintWithDoc) printToCSV(data []printData) {
+	sort.Slice(data, func(i, j int) bool {
+		return data[i].filmDate.Before(data[j].filmDate)
+	})
+
+	lines := []string{"Unit UID", "Original Name", "film date", "topic name"}
+	for _, d := range data {
+		l := fmt.Sprintf("\n%s, %s, %s, %s", d.cuUid, strings.ReplaceAll(d.oName, ",", "->"), d.filmDate.String(), d.topic)
+		lines = append(lines, l)
+	}
+	b := []byte(strings.Join(lines, ","))
+	p := path.Join(viper.GetString("source-import.results-dir"), "move-kitvei-makor.csv")
+	err := ioutil.WriteFile(p, b, 0644)
+	utils.Must(err)
+}
+
+func (c *PrintWithDoc) openDB() *sql.DB {
+	mdb, err := sql.Open("postgres", viper.GetString("source-import.test-url"))
+	utils.Must(err)
+	utils.Must(mdb.Ping())
+	boil.SetDB(mdb)
+	boil.DebugMode = true
+	utils.Must(common.InitTypeRegistries(mdb))
+	return mdb
+}
+
+func FindOrigin(mdb *sql.DB, id int64) (*models.ContentUnit, error) {
```

Comment:
In both places you call this function you have a unit.
Consider loading it's associations with the ORM.
Something like this https://github.com/Bnei-Baruch/mdb/blob/master/api/handlers.go#L734